src/transaction/fee.rs

```diff
@@ -75,6 +75,20 @@ impl TransactionFee {
             .map(|(bytes, total)| Self::new(bytes, total))
     }
 
+    /// Attempt to calculate a gas as asset value, using the price factor defined in the consensus
+    /// parameters.
+    ///
+    /// Will return `None` if overflow occurs
+    pub fn gas_to_value(params: &ConsensusParameters, gas: Word, price: Word) -> Option<Word> {
```

Comment:
[nit] maybe call this gas_refund_value, since we only floor the rounding when calculating the refund. When doing initial balances, we do ceiling based rounding.